lib/cucloud/rds_utils.rb

```diff
@@ -13,6 +13,67 @@ def get_instance(db_instance_identifier)
       resource.db_instance(db_instance_identifier)
     end
 
+    # Determine if a givne db instance exist
+    # @param db_instance_identifier [String] RDS instance identifier
+    # @return [boolean]
+    def does_db_exist?(db_instance_identifier)
+      get_instance(db_instance_identifier).instance_create_time
+      true
+    rescue Aws::RDS::Errors::DBInstanceNotFound
+      false
+    end
+
+    # Delete a givne db instance
+    # @param db_instance_identifier [String] RDS instance identifier
+    # @param db_snapshot_identifier [String] Name for final snapshot, default is nil
+    def delete_db_instance(db_instance_identifier, db_snapshot_identifier = nil)
+      if does_db_exist?(db_instance_identifier)
+        if db_snapshot_identifier.nil?
+          @rds.delete_db_instance(db_instance_identifier: db_instance_identifier, skip_final_snapshot: true)
+        else
+          @rds.delete_db_instance(db_instance_identifier: db_instance_identifier,
+                                  final_db_snapshot_identifier: db_snapshot_identifier)
+        end
+
+        @rds.wait_until(:db_instance_deleted, db_instance_identifier: db_instance_identifier)
+      else
+        raise Aws::RDS::Errors::DBInstanceNotFound.new(db_instance_identifier, '')
+      end
+    end
+
+    # Restore DB from a snapshot
+    # @param db_instance_identifier [String] RDS instance identifier
+    # @param db_snapshot_identifier [String] Name for final snapshot, default is nil
+    def restore_db(db_instance_identifier, restore_from, options = {})
+      unless does_db_exist?(db_instance_identifier)
+        db_snapshot_identifier =
+          options[:db_snapshot_identifier].nil? ? find_latest_snapshot(restore_from) : options[:db_snapshot_identifier]
+        options[:db_instance_identifier] = db_instance_identifier
+        options[:db_snapshot_identifier] = db_snapshot_identifier
+        @rds.restore_db_instance_from_db_snapshot(options)
+      end
+    end
+
+    # Delete a givne db instance
+    # @param db_instance_identifier [String] RDS instance identifier
+    # @return [String] Most recent snapshot ID for given RDS instance
+    def find_latest_snapshot(db_instance_identifier)
+      latest_snapshot_time = Time.new(2002)
+      latest_snap_shot = nil
+      snapshots_info = @rds.describe_db_snapshots(db_instance_identifier: db_instance_identifier)[:db_snapshots]
```

Comment:
Might need to filter to manual snapshots here, otherwise need to handle paging...